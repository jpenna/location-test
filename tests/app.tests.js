/* eslint import/no-extraneous-dependencies: "off" */
/*  eslint import/no-unresolved: "off" */

import React from 'react';
import { shallow } from 'enzyme';

import App from 'src/app';
import Title from 'src/components/title';

describe('<App />', () => {
  let shallowed;

  beforeEach(() => {
    shallowed = shallow(<App />);
  });

  it('should have <Title /> component', () => {
    // console.log(shallowed.html());
    expect(shallowed.contains(<Title />)).toBeTruthy();
  });

  it('should have <LocationData /> component', () => {
    expect(shallowed.find('LocationData').length).toEqual(2);
  });

  it('should have <InputUrl /> component', () => {
    expect(shallowed.find('InputUrl').length).toEqual(1);
  });

  it('should set URL on input component submit', () => {
    const url = 'myurl.com';
    shallowed.instance().handleUrlSubmit(url);
    expect(shallowed.state('url')).toEqual(url);
  });
});
