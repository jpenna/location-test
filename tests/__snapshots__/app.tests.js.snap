// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly 1`] = `
<div
  className="section"
>
  <div
    className="container"
  >
    <div
      className="margin-bottom-2"
    >
      <div
        className="has-text-centered"
      >
        <h1
          className="title is-1"
        >
          GeoLocation Test
        </h1>
        <h2
          className="subtitle has-margin-bottom-10"
        >
          Find you and a website
        </h2>
      </div>
    </div>
    <div
      className="margin-bottom-2"
    >
      <div
        className="content"
      >
        <form
          className="has-text-centered"
          onSubmit={[Function]}
        >
          <div
            className="field has-addons has-addons-centered"
          >
            <div
              className="control"
            >
              <input
                className="input"
                onChange={[Function]}
                pattern="^[-a-zA-Z0-9@:%._\\\\+~#=]{2,256}\\\\.[a-z]{2,6}\\\\b([-a-zA-Z0-9@:%_\\\\+.~#?&=]*)"
                placeholder="website.com"
                type="text"
                value=""
              />
              <div
                className="help is-danger show-invalid"
              >
                Format: www.google.com / google.com
              </div>
            </div>
            <div
              className="control"
            >
              <button
                className="button is-primary"
                type="submit"
              >
                Locate!
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
    <div
      className="is-flex root-columns"
    >
      <div
        className="column is-4 user-column"
      >
        <div>
          <div>
            <div
              className="is-relative"
            >
              <div
                className="notification is-primary info-balloon is-hidden"
              >
                <button
                  className="delete"
                  onClick={[Function]}
                />
                This is your data according to freejeoip.net at 
                
              </div>
              <button
                className="js-showInfo is-buttonless "
                disabled={true}
                onClick={[Function]}
              >
                <h3
                  className="title is-5 is-inline-block"
                >
                  Your Location
                </h3>
                <span
                  className="icon info-sign is-hidden"
                >
                  <i
                    className="fa fa-info-circle"
                  />
                </span>
              </button>
              <span
                className="field"
                hidden={true}
              >
                <button
                  className="reset-button button is-small is-borderless is-danger inverted text-bottom-align"
                  onClick={[Function]}
                  type="button"
                >
                  <span
                    className="icon"
                  >
                    <i
                      className="fa fa-trash"
                    />
                  </span>
                </button>
              </span>
            </div>
          </div>
          <div
            className="section mobile-paddingless"
            hidden={0}
          >
            <button
              className="get-location-button is-buttonless pointer"
              onClick={[Function]}
            >
              <div
                className="icon button is-info is-rounded margin-bottom-half"
              >
                <i
                  className="fa fa-map-marker"
                />
              </div>
              <div
                className="title is-6"
              >
                My Location
              </div>
            </button>
          </div>
          <div
            className="section mobile-paddingless"
            hidden={true}
          >
            <div
              className="content has-text-centered"
              hidden=""
            >
              Let's get some page info?
              <br />
              Type in a URL above
            </div>
          </div>
          <div
            className="is-danger has-text-centered help title is-6 has-new-line is-marginless"
          >
            
          </div>
          <div
            hidden=""
          />
        </div>
      </div>
      <div
        className="column is-4 map-column"
      >
        <div>
          <div
            className="map-frame"
            id="map"
          />
          <div />
        </div>
      </div>
      <div
        className="column is-4 web-column"
      >
        <div>
          <div>
            <div
              className="is-relative"
            >
              <div
                className="notification is-primary info-balloon is-hidden"
              >
                <button
                  className="delete"
                  onClick={[Function]}
                />
                This is your data according to freejeoip.net at 
                
              </div>
              <button
                className="js-showInfo is-buttonless "
                disabled={true}
                onClick={[Function]}
              >
                <h3
                  className="title is-5 is-inline-block"
                >
                  Webpage Location
                </h3>
                <span
                  className="icon info-sign is-hidden"
                >
                  <i
                    className="fa fa-info-circle"
                  />
                </span>
              </button>
              <span
                className="field"
                hidden={true}
              >
                <button
                  className="reset-button button is-small is-borderless is-danger inverted text-bottom-align"
                  onClick={[Function]}
                  type="button"
                >
                  <span
                    className="icon"
                  >
                    <i
                      className="fa fa-trash"
                    />
                  </span>
                </button>
              </span>
            </div>
          </div>
          <div
            className="section mobile-paddingless"
            hidden={true}
          >
            <button
              className="get-location-button is-buttonless pointer"
              onClick={[Function]}
            >
              <div
                className="icon button is-info is-rounded margin-bottom-half"
              >
                <i
                  className="fa fa-map-marker"
                />
              </div>
              <div
                className="title is-6"
              >
                My Location
              </div>
            </button>
          </div>
          <div
            className="section mobile-paddingless"
            hidden={0}
          >
            <div
              className="content has-text-centered"
              hidden=""
            >
              Let's get some page info?
              <br />
              Type in a URL above
            </div>
          </div>
          <div
            className="is-danger has-text-centered help title is-6 has-new-line is-marginless"
          >
            
          </div>
          <div
            hidden=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
